isolate.go

```diff
@@ -75,6 +77,53 @@ func (i *Isolate) IsExecutionTerminating() bool {
 	return C.IsolateIsExecutionTerminating(i.ptr) == 1
 }
 
+type CompileOptions struct {
+	CachedData *ScriptCompilerCachedData
+
+	Option ScriptCompilerCompileOption
+}
+
+// CompileUnboundScript will create an UnboundScript (i.e. context-indepdent)
+// using the provided source JavaScript, origin (a.k.a. filename), and options.
+// If options contain a non-null CachedData, compilation of the script will use
+// that code cache.
+// error will be of type `JSError` if not nil.
+func (i *Isolate) CompileUnboundScript(source, origin string, opts CompileOptions) (*UnboundScript, error) {
+	cSource := C.CString(source)
+	cOrigin := C.CString(origin)
+	defer C.free(unsafe.Pointer(cSource))
+	defer C.free(unsafe.Pointer(cOrigin))
+
+	if opts.CachedData != nil {
+		opts.Option = scriptCompilerConsumeCodeCache
+	}
+
+	cOptions := C.CompileOptions{
+		compileOption: C.int(opts.Option),
+	}
+	if opts.CachedData != nil {
+		cOptions.cachedData = C.ScriptCompilerCachedData{
+			data:   (*C.uchar)(unsafe.Pointer(&opts.CachedData.Bytes[0])),
+			length: C.int(len(opts.CachedData.Bytes)),
+		}
+		if opts.CachedData.Rejected {
+			cOptions.cachedData.rejected = C.int(1)
```

Comment:
`C.IsolateCompileUnboundScript` doesn't actually use the incoming `cOptions.cachedData.rejected` field because whether it was rejected before or not by some other usage is irrelevant to whether it will be useful to this call so this can probably be removed.